kivy/core/text/markup.py

```diff
@@ -752,7 +752,14 @@ def p_restricted(line, uw, c):
         s = self.get_extents(last_text)
         if len(last_text):
             line1.append(LayoutWord(last_word.options, s[0], s[1], last_text))
-        elps.options = last_word.options
```

Comment:
I think this should have been `elps.options = old_opts`, which is basically redundant since the ellipse options had been set properly earlier. It would be good to test to see if the ellipse ever takes the opts of the word to the left, because that would be wrong then and this line should just be removed.